examples/action_dann_lightn/model.py

```diff
@@ -60,49 +64,202 @@ def get_feat_extractor(model_name, num_classes, num_channels):
 
     Args:
         model_name: The name of the feature extractor.
-        num_classes: The class number for the specific setting. (Default: No use)
-        num_channels: The number of image channels. (Default: No use, may used in RGB & Flow)
+        image_modality: Image type. (RGB or Optical Flow)
+        attention: The attention type.
+        num_classes: The class number of specific dataset. (Default: No use)
 
     Returns:
         feature_network: The network to extract features.
-        feature_dim: The dimension of the feature network output. It is a convention when
-                    the input dimension and the network is fixed.
+        class_feature_dim: The dimension of the feature network output for ClassNet.
+                        It is a convention when the input dimension and the network is fixed.
+        dmn_feature_dim: The dimension of the feature network output for DomainNet.
     """
+    attention_list = ["SELayerC", "SELayerT", "SELayerCoC", "SELayerMC", "SELayerCT", "SELayerTC", "SELayerMAC"]
 
-    if model_name == "I3D":
-        pretrained_model = "rgb_imagenet" if num_channels == 3 else "flow_imagenet"
-        feature_network = i3d(name=pretrained_model, num_channels=num_channels, pretrained=True)
-        # model.replace_logits(num_classes)
-        feature_dim = 1024
-    elif model_name == "R3D_18":
-        feature_network = r3d_18(pretrained=True)
-        feature_dim = 512
-    elif model_name == "R2PLUS1D_18":
-        feature_network = r2plus1d_18(pretrained=True)
-        feature_dim = 512
-    elif model_name == "MC3_18":
-        feature_network = mc3_18(pretrained=True)
-        feature_dim = 512
+    if attention in attention_list:
+        att = True
+    elif attention == "None":
+        att = False
     else:
-        raise ValueError("Unsupported model: {}".format(model_name))
-    return feature_network, feature_dim
+        raise ValueError("Wrong attention. Current: {}".format(attention))
+
+    if image_modality == "rgb":
+        if model_name == "I3D":
+            pretrained_model = "rgb_imagenet"
+            if att:
+                logging.info("{} using {}".format(model_name, attention))
+                feature_network = se_i3d_joint(
+                    rgb_pt=pretrained_model, flow_pt=None, num_classes=num_classes, attention=attention, pretrained=True
+                )
+            else:
+                logging.info("No SELayer.")
+                feature_network = i3d_joint(
+                    rgb_pt=pretrained_model, flow_pt=None, num_classes=num_classes, pretrained=True
+                )
+            # model.replace_logits(num_classes)
+            class_feature_dim = 1024
+            dmn_feature_dim = class_feature_dim
+
+        elif model_name == "R3D_18":
+            if att:
+                logging.info("{} using {}".format(model_name, attention))
```

Comment:
Lots of repetition here, above and below few lines. 
Consider to define a function to reduce such redundancy.